csum-file.h

```diff
@@ -16,7 +16,9 @@ struct hashfile {
 	const char *name;
 	int do_crc;
 	uint32_t crc32;
-	unsigned char buffer[8192];
+	size_t buffer_len;
+	unsigned char *buffer;
+	unsigned char *check_buffer;
```

Comment:
An alternative would be to use the `FLEX*` macros, to avoid allocating a `struct hashfile` and _then_ a `buffer`.

Reply:
Not every instance needs a `check_buffer`. The `buffer` change was to prevent stack size problems as the struct grows in size.